System.MissingMethodException when an abstract class inherits class which implements IAssemblyContext

When AssemblyRunner invokes OnAssemblyStart via reflection on an abstract class instance an exception is thrown. Only concrete implementations of IAssemblyContext should have its OnAssemblyStart method invoked

#[error]Error: Machine Specifications Visual Studio Test Adapter - Fatal error while executing test.
 ##[error]System.MissingMethodException: Cannot create an abstract class.
 ##[error] at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
 ##[error] at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
 ##[error] at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
 ##[error] at System.Activator.CreateInstance(Type type, Boolean nonPublic)
 ##[error] at System.Activator.CreateInstance(Type type)
 ##[error] at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()

--- a/src/Machine.Specifications/Explorers/AssemblyExplorer.cs
+++ b/src/Machine.Specifications/Explorers/AssemblyExplorer.cs
@@ -48,7 +48,7 @@ namespace Machine.Specifications.Explorers
         public IEnumerable<IAssemblyContext> FindAssemblyContextsIn(Assembly assembly)
         {
             return assembly.GetExportedTypes()
-              .Where(x => x.GetTypeInfo().IsClass && x.GetTypeInfo().IsAbstract == false && x.GetInterfaces().Contains(typeof(IAssemblyContext)))
+              .Where(x => x.GetTypeInfo().IsClass && !x.GetTypeInfo().IsAbstract && x.GetInterfaces().Contains(typeof(IAssemblyContext)))
               .Select(x => (IAssemblyContext)Activator.CreateInstance(x));
         }
 

--- a/src/Machine.Specifications/Explorers/AssemblyExplorer.cs
+++ b/src/Machine.Specifications/Explorers/AssemblyExplorer.cs
@@ -48,7 +48,7 @@ namespace Machine.Specifications.Explorers
         public IEnumerable<IAssemblyContext> FindAssemblyContextsIn(Assembly assembly)
         {
             return assembly.GetExportedTypes()
-              .Where(x => x.GetInterfaces().Contains(typeof(IAssemblyContext)))
+              .Where(x => x.GetTypeInfo().IsClass && x.GetTypeInfo().IsAbstract == false && x.GetInterfaces().Contains(typeof(IAssemblyContext)))
               .Select(x => (IAssemblyContext)Activator.CreateInstance(x));
         }
 

